add support for api/v2 endpoints, enable retry

alpinweis opened the pull request:
- support multiple versions of the API
- add methods for api/v2 endpoints: pipeline/:id, pipeline/:id/config, workflow/:id, workflow/:id/jobs, project/:slug
- enable retrying for failed requests (5xx)
- add `shallow` param to get recent builds

levlaz commented on .gitignore:
```diff
@@ -36,6 +36,7 @@ pip-log.txt
 pip-delete-this-directory.txt
 
 # Unit test / coverage reports
+circleci_api_py.html
```
shouldn't this just be a part of htmlcov?

alpinweis replied:
for me it gets created in the package root folder. I just moved it a few lines below if that's what you mean

alpinweis committed:
add support for api/v2 endpoints, enable retry

- support multiple versions of the API
- add methods for api/v2 endpoints: pipeline/:id, pipeline/:id/config, workflow/:id, workflow/:id/jobs, project/:slug
- enable retrying for failed requests (5xx)

prasket commented:
Looking forward to this PR. Just found this project and was sad to see it didnt support v2 until I saw this. :)

alpinweis commented:
any interest in reviewing and moving forward with this PR ?

levlaz commented:
@alpinweis I really appreciate your patience here. I am going to review this again and hopefully get it merged in this week.

alpinweis committed:
add 'shallow' param to get recent builds, more updates
